internal/addon/manifests/charts/mcoa/templates/cluster-role.yaml

```diff
@@ -15,4 +15,7 @@ rules:
   - apiGroups: ["operators.coreos.com"]
     resources: ["operatorgroups"]
     verbs: ["get", "list", "watch", "create", "update", "delete", "patch"]
+  - apiGroups: ["monitoring.coreos.com"]
+    resources: ["prometheusagents", "prometheusrules", "scrapeconfigs"]
+    verbs: ["get", "list", "watch", "create", "update", "delete", "patch"]
```

Comment:
I believe this might not be necessary as the work-agent already has cluster-admin like permissions.

Reply:
humm, I remember I had issues with that, but maybe in the end it was fixed by something else. I'll check this again.

Reply:
Just tested again, and it is needed. I get the following log line from the spoke `klusterlet-agent` :
```
AppliedManifestWorkController" controller failed to sync "addon-multicluster-observability-addon-deploy-0", err: [failed to get resource monitoring.coreos.com/v1alpha1, Resource=scrapeconfigs with key open-cluster-management-agent-addon/platform-metrics-default: scrapeconfigs.monitoring.coreos.com "platform-metrics-default" is forbidden: User "system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa" cannot get resource "scrapeconfigs" in API group "monitoring.coreos.com" in the namespace "open-cluster-management-agent-addon", failed to get resource monitoring.coreos.com/v1alpha1, Resource=prometheusagents with key open-cluster-management-agent-addon/acm-user-workload-metrics-collector: prometheusagents.monitoring.coreos.com "acm-user-workload-metrics-collector" is forbidden: User "system:serviceaccount:open-cluster-management-agent:klusterlet-work-sa" cannot get resource "prometheusagents" in API group "monitoring.coreos.com" in the name...
```